contracts/serverless-v2/RelaySubscriptions.sol

```diff
@@ -257,67 +274,75 @@ contract RelaySubscriptions is
     /// @notice Error for when the job subscription owner tries to terminate the subscription 
     ///         before the termination condition is reached.
     error RelaySubscriptionsTerminationConditionPending();
+    /// @notice Thrown when the provided execution environment is not supported globally.
+    error RelaySubscriptionsEnvUnsupported();
+
+    modifier isValidEnv(uint8 _env) {
+        if(!RELAY.isEnvSupported(_env))
+            revert RelaySubscriptionsEnvUnsupported();
+        _;
+    }
 
     //-------------------------------- internal functions start --------------------------------//
 
     function _startJobSubscription(
-        bytes32 _codehash,
-        bytes calldata _codeInputs,
-        uint256 _userTimeout,
-        uint256 _maxGasPrice,
-        address _refundAccount,
-        address _callbackContract,
-        uint256 _callbackGasLimit,
-        uint256 _periodicGap,
-        uint256 _usdcDeposit,
-        uint256 _startTimestamp,
-        uint256 _terminationTimestamp,
+        JobSubscriptionParams memory _jobSubsParams,
```

Comment:
Nope, since we are updating the startTime here - 
```
if(_jobSubsParams.startTime < block.timestamp)
    _jobSubsParams.startTime = block.timestamp;
```